perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -903,6 +903,28 @@ sub report_edit : Path('report_edit') : Args(1) {
 
         if ( $problem->state ne $old_state ) {
             $c->forward( 'log_edit', [ $id, 'problem', 'state_change' ] );
+
+            unless (
+              $problem->state eq 'unconfirmed' || $old_state eq 'unconfirmed' ||
+              $problem->state eq 'hidden' || $old_state eq 'hidden'
```

Comment:
If the old state was hidden, do we want an update to record it's no longer hidden?
Perhaps even if new state is hidden too, just so it's all recorded (and if it's unhidden later, you have a log of when it was hidden/unhidden then).
Not sure about unconfirmed, but I guess it even could do then as well (doubtful that's ever used anyway).

Reply:
I wasn't sure about exposing hiding, unhiding and confirming reports on the report page which is why this isn't here. I suppose I could record this here and change the logic on the report page, although that might be a bit fiddly. Possibly a non-public flag in the extra information is the way to go.

Reply:
I think I lean towards recording all state changes for consistency and so we can always recreate the full history, and if we don't want to display ones, do that on display as you say.